cache/s3proxy/s3proxy.go

```diff
@@ -114,9 +114,10 @@ func New(s3Config *config.S3CloudStorageConfig, accessLogger cache.Logger,
 
 func (c *s3Cache) objectKey(hash string, kind cache.EntryKind) string {
 	if c.prefix == "" {
-		return fmt.Sprintf("%s/%s", kind, hash)
+		return cache.Key(kind, hash)
 	}
-	return fmt.Sprintf("%s/%s/%s", c.prefix, kind, hash)
+
+	return fmt.Sprintf("%s/%s", c.prefix, cache.Key(kind, hash))
```

Comment:
I think adding a flag to enable the new key format is better than the alternative. I am unsure how (or if) people are removing old s3 blobs, and that might setup might also require an update when using the new key format.

Do you have any ideas for a good name? https://cloud.google.com/storage/quotas suggests that GCS doesn't rate limit any operations we need to scale, so keeping this as an s3-specific setting seems wise. Maybe `--s3.new_key_format` or `--s3.three_level_key`?

Reply:
Yeah, I think since GCS is a natively supported cache, rate limiting should not be a concern there.  Im happy with either of those, first makes more sense personally.